akka-http/src/main/scala/akka/http/javadsl/server/RequestContext.scala

```diff
@@ -50,6 +58,14 @@ class RequestContext private (val delegate: scaladsl.server.RequestContext) {
       .fast.map(r ⇒ r.asJava: RouteResult)(akka.dispatch.ExecutionContexts.sameThreadExecutionContext).toJava
   }
 
+  def redirect(uri: Uri, redirectionType: StatusCode): CompletionStage[RouteResult] = {
+    import akka.http.javadsl.model._
+    import akka.http.javadsl.server.Marshaller
+    complete(HttpResponse.create().withStatus(redirectionType).addHeader(Location.create(uri)), Marshaller.opaque(new JFunction[HttpResponse, HttpResponse] {
```

Comment:
We are unable to use `javaIdentityFunction` because it is not a `java.util.function.Function[?,?]`. 

Delegate in on the move.